### PR TITLE
Impl `AvlTree` properties

### LINKED ISSUE #425: [300 SigmaUSD] AvlTree properties
Implement `SMethod`(in `types::savltree`) and evaluation (in `eval::savltree`) of the following properties and methods:
- digest
- enabledOperations
- keyLength
- valueLengthOpt
- isInsertAllowed
- isUpdateAllowed
- isRemoveAllowed
- updateOperations
- updateDigest

Reference Scala implementation - https://github.com/ScorexFoundation/sigmastate-interpreter/blob/c863a9b1a82589e47b15f76f3affdb30a475e740/sigmastate/src/main/scala/sigmastate/types.scala#L2018-L2101
and
https://github.com/ScorexFoundation/sigmastate-interpreter/blob/c863a9b1a82589e47b15f76f3affdb30a475e740/sigmastate/src/main/scala/sigmastate/eval/CostingDataContext.scala#L128-L150

### PR DESCRIPTION
**Pull request by kettlebell**
Close #425

### CODE REVIEW

File: ergotree-interpreter/src/eval/savltree.rs [hunk 1]
```diff
@@ -189,6 +265,161 @@ mod tests {
             unreachable!();
         }
     }
+    proptest! {
+        #[test]
+        fn eval_avl_properties(v in any::<AvlTreeData>(), new_ops in any::<AvlTreeFlags>(), new_digest in any::<ADDigest>()) {
```
Comment (greenhat) [inline ergotree-interpreter/src/eval/savltree.rs:270]: If this test failed it'd be hard to pinpoint the failing case, especially looking at the logs on CI. Consider splitting it into separate tests or providing an error message to identify which case is failed.

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking good! Please check my comments.

File: ergotree-interpreter/src/eval/savltree.rs [hunk 2]
```diff
@@ -15,6 +20,76 @@ use scorex_crypto_avltree::operation::Operation;
...
+    Ok(Value::AvlTree(Box::new(avl_tree_data)))
+};
+
+pub(crate) static UPDATE_DIGEST_EVAL_FN: EvalFn = |_env, _ctx, obj, args| {
+    let mut avl_tree_data = obj.try_extract_into::<AvlTreeData>()?;
+    let new_digest = {
+        let v = args.get(0).cloned().ok_or_else(|| {
+            EvalError::AvlTree("eval is missing first arg (new_digest)".to_string())
+        })?;
+        let bytes: [u8; 33] = v
```
[the remark this answers is not in the document]
Comment (kettlebell) [reply ergotree-interpreter/src/eval/savltree.rs]: Done.

File: ergotree-interpreter/src/eval/savltree.rs (hunk 1 above)
Comment (kettlebell) [reply ergotree-interpreter/src/eval/savltree.rs]: I split it into separate tests.

**Review by greenhat — APPROVED** [review]
Great! Thank you!

**Pull request merged**